docs/source/using/managing-minishift.adoc

```diff
@@ -322,3 +322,57 @@ Enables provisioning the OpenShift link:https://docs.openshift.org/latest/archit
 
 `extra-clusterup-flags`::
 Enables passing flags that are not directly exposed in the Minishift CLI directly to `oc cluster up`.
+
+
+[[hyperv-static-ip]]
+=== Assign IP address to Hyper-V (experimental)
+
+Since the Internal Virtual Switch for Hyper-V does not provide a DHCP offer option, an IP address needs to be provided in a different way.
+For Hyper-V a functionality is provided to assign an IP address on startup using the Data Exchange Service.
+
+[IMPORTANT]
+====
+-  This only works with the CentOS/RHEL based image in combination with Hyper-V
+====
+
+To make this work you need to create a Virtual Switch using NAT
+
+- link:https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/setup-nat-network[Set up a NAT network]
+
+
+[NOTE]
+====
+WinNAT is limited to one NAT network per host. For more details about capabilities, and limitations, please see the link:https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/setup-nat-network[WinNAT capabilities and limitations blog]
+====
+
+The following command will attempt to assign an IP address for use on the Internal Virtual Switch 'MyInternal'.
+
+----
+PS> $env:MINISHIFT_ENABLE_EXPERIMENTAL="y"
+PS> $env:HYPERV_VIRTUAL_SWITCH="MyInternal"
+PS> minishift.exe start `
+  --iso-url centos `
+  --network-ipaddress 192.168.1.10 `
```

Comment:
So these options appear in the start command, even though they only apply for Hyper-V? Would it not be nicer to have a more generic approach.  I am concerend about flag proliferation and also people trying to use these flags with the wrong driver?

If there is no generic mechanism, we could while it is experimental go with env settings only. This is how driver specific options are set. So why could an initial version of this not be based on some HYPERV_XYZ options? 

We already had some bad experience having a VirtuaBox specific flag once upon a time.

Reply:
Why prefix, as they will also work for other hypervisors in the follow up issue... just not on startup.

they are currently hidden behind ENABLE_EXPERIMENTAL, but can be enforced with `runtime.GOOS == "windows"`

Reply:
On startup, you will likely only use `--network-ipaddress` as the remaining values are network-specific, so you will likely store them as

```
m config set network-gateway 10.0.75.1
m config set network-netmask 24
m config set network-nameserver 208.67.220.220
```
anyways

Reply:
If hypervisor or platform does not support this, the following is shown:

```
PS C:\work\src\github.com\minishift\minishift\out\windows-amd64> .\minishift.exe start --vm-driver virtualbox --network-ipaddress 10.0.21.128
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB
-- Attempting to set network settings ... FAIL
   not supported on this platform or hypervisor
-- Starting Minishift VM ... 
```